internal/terminal/log.go

```diff
@@ -0,0 +1,117 @@
+package terminal
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/iancoleman/orderedmap"
+)
+
+// LogLevel is the level of a terminal log
+type LogLevel string
+
+// set of supported log levels
+const (
+	LogLevelInfo  LogLevel = "info"
+	LogLevelError LogLevel = "error"
+)
+
+var (
+	allLogLevels = []LogLevel{LogLevelInfo, LogLevelError}
+
+	longestLogLevel = func() LogLevel {
+		var longest LogLevel
+		var longestLength int
+		for _, level := range allLogLevels {
+			if l := len(level); l > longestLength {
```

Comment:
[nit] could get rid of `longestLength`/avoid having to keep track of two vars here by just checking `l > len(longest)`. 🤷 there are only two log levels currently and this will only run once

Reply:
good enough for me (at best, there may be an added `DEBUG` but I can't imagine any more)